### ISSUE TITLE
`single_char_pattern` broken since rust 1.80

### ISSUE THREAD

**Issue by abezukor** [rust-lang/rust-clippy#13954]
### Summary

As far as I can tell `clippy::single_char_pattern` has been broken since clippy 1.80. The last version that throws a warning is clippy 1.79. I dont see anything in [the changelog](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-180) that indicates this was intentional.

### Lint Name

single_char_pattern

### Reproducer

I tried this code:

```rust
fn main() {
    let to_split: &str = "foo, bar, baz";
    let _split = to_split.split_once(",");
}
```
I expected to see this happen:
```
warning: single-character string constant used as pattern
 --> src/main.rs:3:38
  |
3 |     let _split = to_split.split_once(",");
  |                                      ^^^ help: consider using a `char`: `','`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_pattern
  = note: `#[warn(clippy::single_char_pattern)]` on by default
```
Instead, this happened:
```
cargo clippy   
    Checking lint_test v0.1.0 (/tmp/lint_test)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.05s
```

### Version

```text
rustc 1.86.0-nightly (b3b368a18 2025-01-05)
binary: rustc
commit-hash: b3b368a1833a26f5d48f51c45159f46e6cd01711
commit-date: 2025-01-05
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.6
```

**Comment by samueltardieu** [comment]
This lint is in the `pedantic` category. Did you enable it?

**Comment by abezukor** [comment]
Oh never mind, I missed [the note](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#moves-and-deprecations-3) that it got moved to pedantic.
